set-i18n.js

```diff
@@ -0,0 +1,82 @@
+const REPLACE = /^(?:text|placeholder)\s(?:<[^\s]*\s)?(.*)/;
+const GET_TEXT_AND_SLOT = /\/(text|placeholder)\s(<[^\s]*\s)?/;
+
+const isText = line => /^(text|placeholder)/.test(line);
+const isBlock = line => /^[A-Z]/.test(line);
+const replaceText = (line, newText) =>
+  line.replace(line.match(REPLACE)[1], newText);
+const getPath = key => {
+  const match = key.match(/([^\s]+)\s/);
+  return match[1].split('/');
+};
+const getStartIndex = (path, lines) => {
+  let startIndex = 0;
+  path.forEach((name, index) => {
+    for (let i = startIndex; i < lines.length; i++) {
+      if (isBlock(name) && new RegExp(name).test(lines[i])) {
+        startIndex = i + 1;
+        break;
+      }
+    }
+  });
+  return startIndex;
+};
+
+function setI18n({ source, view, translations }) {
+  const lines = source
+    .replace(/\r\n/g, '\n')
+    .split('\n')
+    .map(line => line.trim());
+  const languages = Object.keys(translations);
+  const keys = Object.keys(translations['en']);
+
+  keys.forEach(key => {
+    const startIndex = getStartIndex(getPath(key), lines);
+    languages.forEach(lang => {
+      var added = false;
+      let i = startIndex;
+      while (!added) {
+        if (lang === 'en') {
```

Comment:
Should be writing back the default language's string? It feels to me like we would want to skip that language altogether in line 30 `const languages = Object.keys(translations).filter(language => language !== defaultLanguage);`

Reply:
I am updating it here because I presume default language strings might get updated through these translation files.
If the source of truth for the default language is the view file, then I'll filter it out. If the source of truth is the translations file, then I'd keep it as is.

Reply:
I see, yeah, the source of truth is the view file :)